File: Warehouse.Picking/Domain/Article.cs
using System;
using System.Collections.Generic;

namespace warehouse.picking.api.Domain
{
    [Serializable]
    public class Article
    {
        public int Id { get; set; }
        
        public string NoteId { get; set; }
        
        public string Name { get; set; }

        public string Gtin { get; set; }

        public ArticleQuantity Quantity { get; set; }

        public ArticleBundle Bundle { get; set; }

        public void UpdateProcessedQuantity(int q)
        {
            Quantity.Processed += q;
        }

        public bool IsUnfinished()
        {
            return Quantity.Processed < Quantity.Expected;
        }

        public Article(int id, string noteId, string name, string gtin, ArticleQuantity quantity, ArticleBundle bundle)
        {
            Id = id;
            NoteId = noteId;
            Name = name;
            Gtin = gtin;
            Quantity = quantity;
            Bundle = bundle;
        }
    }

    [Serializable]
    public class Articles
    {
        public Articles(List<Article> articles)
        {
            Items = articles;
        }

        public List<Article> Items { get; }
        
    }
}

File: android/app/src/main/java/com/eazzyapps/example/android/Article.kt
package com.eazzyapps.example.android

data class Articles(
    val items: List<Article>
)

data class Article(
    val id: Int,
    val noteId: String,
    val name: String,
    val gtin: String,
    val quantity: ArticleQuantity,
    val bundle: ArticleBundle
)

data class ArticleQuantity(
    val expected: Int,
    val processed: Int
)

data class ArticleBundle(
    val id: Int,
    val name: String
)